Improve `react-big-calendar` UI & UX

diogogmatos opened the pull request:
Redesigned the whole `react-big-calendar` component to a more modern, appealing and user-friendly look. Most of the changes are grounded on the override of the rbc styling in `globals.css`, although some others were done via tweaks in the `<Calendar />` properties.

This included redesigning of the Week, Month and Schedule view. I took inspiration from TailwindUI's Calendar components.

**Preview:**

Month View
![image](https://github.com/cesium/calendarium/assets/80540164/416a059c-dbe8-4ed9-89bb-647fc77fc80a)

Week View
![image](https://github.com/cesium/calendarium/assets/80540164/f6682711-5cdc-4925-9e03-3b910c71cb2a)

Schedule View
![image](https://github.com/cesium/calendarium/assets/80540164/a5bfc9fb-32a5-445e-a1d8-a1a18350812c)

Toolbar
![Screenshot_20230619_224114](https://github.com/cesium/calendarium/assets/80540164/d062b409-f36e-4b48-a73c-4f28a8bc4d28)

**Comparisons (Before/After):**

Month View
![image](https://github.com/cesium/calendarium/assets/80540164/d4751f22-73a7-4a06-b59f-e469302309f9)

Week View
![image](https://github.com/cesium/calendarium/assets/80540164/7b1bae87-f857-4f75-b2b1-add4805e8f6b)

Schedule View
![image](https://github.com/cesium/calendarium/assets/80540164/62fc3465-ad8d-4a5e-9ff1-5f462556af0c)

RuiL1904 commented:
Clicking on "+2 more" is not working for me. Is it working for you, @diogogmatos?
![image](https://github.com/cesium/calendarium/assets/76881129/6439241a-cbf3-42da-b469-b2c753ffeed1)

diogogmatos commented:
> Clicking on "+2 more" is not working for me. Is it working for you, @diogogmatos?
> 
> ![image](https://github.com/cesium/calendarium/assets/76881129/6439241a-cbf3-42da-b469-b2c753ffeed1)
> 
> 

soo, the "+x more" button doesn't work because I think it would normally take you to the day view and I disabled the day view. I could try to make it redirect to the Week view instead.

RuiL1904 commented:
Make it redirect to the week view, then. @diogogmatos

diogogmatos commented:
I can also make it redirect to day view but not display the day view button on the toolbar 👀

RuiL1904 commented:
As you want, tbh. Ideally, it would have an arrow side-by-side with the "2+ more" that, when clicked, would show the other two events of the given day (the current ones would disappear - just like it has two pages). But I don't know if that's possible regarding we are using rbc 😅
@diogogmatos

diogogmatos commented:
yeah that's not possible ahaha, but ig i'll make it day view ;) ALSO, just made a discovery. apparently rbc considers an event "all day" when it surpasses midnight, showing it in that space at the top instead of where the hours show (in week or day view). since when we have an all day event we put the hour has "00:00 -> 23:59" it doesn't register as an all day event and just occupies the whole length of the view. however, we can make it "00:00 -> 00:00" and then it will show up as an all day event at the top! leaving the rest of the space to events that already have a defined interval of time and reducing overlaping of events! it looks like this:

![image](https://github.com/cesium/calendarium/assets/80540164/382530b6-6127-48bf-a4c4-80140d7c9f96)

![image](https://github.com/cesium/calendarium/assets/80540164/2891f958-a094-4866-9a0d-36dd94fbec52)

@RuiL1904

RuiL1904 commented:
Yap, looks way nicer.
@diogogmatos

diogogmatos commented:
@RuiL1904 fixed the "+x more" button, converted the events in events.json to support all-day events and updated the Helping Guide to reflect that change.

RuiL1904 commented on HELPING_GUIDE.md:
```diff
@@ -67,6 +67,9 @@ Where `<branch>` is the name of your branch.
 > Check out the `filterId` of each class by searching its name on `shifts.json` or checking the `id` parameter in `filters.json`.
 
+> **Note**
+> For all-day events, `start` and `end` should be equal and composed of: `<event-date> 00:00`. And for multiple-days events: `start: <event-start-date> 00:00` & `end: <event-end-date> 23:59`. Check out the example bellow.
```
```suggestion
> For all-day events, `start` and `end` should be equal and composed of: `<event-date> 00:00`. And for multiple-days events: `start: <event-start-date> 00:00` & `end: <event-end-date> 23:59`. Check out the example below.
```

RuiL1904 reviewed (approved):
This is so much better 😍
